graal-js/src/com.oracle.truffle.js.test.instrumentation/src/com/oracle/truffle/js/test/instrumentation/VarDeclarationsTest.java

```diff
@@ -100,4 +101,20 @@ public void letConstBlock() {
         }).exit();
     }
 
+    @Test
+    public void classDeclare() {
+        evalWithTags("class Foo{}", new Class<?>[]{DeclareTag.class, WriteVariableExpressionTag.class});
+
+        enter(WriteVariableExpressionTag.class, (e1, w1) -> {
+            enter(DeclareTag.class, (e2) -> {
+                assertAttribute(e2, NAME, "Foo");
+                assertAttribute(e2, TYPE, "const");
+            }).exit();
+            enter(WriteVariableExpressionTag.class, (e2, w2) -> {
+                w2.input();
+            }).exit();
+            w1.input();
+        }).exit();
```

Comment:
Any idea why we have 2 write events here?

Reply:
Actually one write here should be internal and hidden for instrumentation. This internal write is created in the parser here: https://github.com/graalvm/graaljs/blob/master/graal-js/src/com.oracle.js.parser/src/com/oracle/js/parser/Parser.java#L1341-L1344

Reply:
The two writes happen in two scopes:
- the outside write is for the global scope that corresponds to the block created here: https://github.com/graalvm/graaljs/blob/master/graal-js/src/com.oracle.js.parser/src/com/oracle/js/parser/Parser.java#L958
- the inner write is for some "internal" block that is created here: https://github.com/graalvm/graaljs/blob/master/graal-js/src/com.oracle.js.parser/src/com/oracle/js/parser/Parser.java#L1532-L1535

The second write and block seem to be unnecessary. Is there any reason to create this extra inner block with the extra write (https://github.com/graalvm/graaljs/blob/master/graal-js/src/com.oracle.js.parser/src/com/oracle/js/parser/Parser.java#L1341-L1344)?